backend/src/openarchiefbeheer/destruction/api/filtersets.py

```diff
@@ -27,6 +30,48 @@ class Meta:
         model = DestructionListItem
         fields = ("destruction_list", "status", "processing_status")
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Add all the filters of the ZaakFilter filterset
+        for name, filter_field in ZaakFilter().filters.items():
+            self.filters["zaak__" + name] = filter_field
```

Comment:
The frontend uses a shared method [getFields](https://github.com/maykinmedia/open-archiefbeheer/blob/main/frontend/src/pages/destructionlist/hooks/useDataGridProps.tsx#L316) that generates the schema for the datagrids. This is the same for all grids, including the one that is present on the destruction list page.

This grid filters on actual zaken, as destruction list items don't exist yet in this context. Therefore, filter fields are not prefixed with "zaak__", as this is a shared function (with good reason) between grids, with various sources of zaak information , it is undesirable to rely on this prefix.

Also: currently the frontend sends QuerySet like expressions for filtering due to leaky abstractions. I'd much rather move away from this format and simplify this, once again: without a zaak__ prefix.

Example of a current filter:

`resultaat__resultaattype__omschrijving__icontains`

could be:

`omschrijving`.

To keep things consistent in my first approach I chained the filters in such way that the QuerySet of the `DestructionListItemFilterset` was limited to destruction list items for zaken in the output of `ZaakFilter`. To get this results all relevant GET parameters passed to DestructionListItemFilterset were passed to ZaakFilter and the queryset was limited to it's output. Resulting in stacked filter that kept backwards compatibly. At the expense of some tidyness. 

I think the approach suggested here relies to much on specifics and is to leaky in its implementation. I'm happy to pick a different approach than my initial PR but we need to keep compatibility and flexibility in mind.

Reply:
Discussed:
- Remove the `zaak__` prefix
- Do the Zaak filters first
- Create an issue for simplifying the names of the zaak filters

Reply:
To discuss further: how to prevent clashes for fields that are present both in the Zaak and the DestructionListItem model